server/rooms.ts

```diff
@@ -668,44 +668,42 @@ export abstract class BasicRoom {
 
 	/**
 	 * @param newID Add this param if the roomid is different from `toID(newTitle)`
+	 * @param moveLogs Whether or not to rename the log file. Defaults to true.
+	 * @param keepAliases Whether or not to point the room's aliases and name to its new name. Defaults to true.
 	 */
-	async rename(newTitle: string, newID?: RoomID) {
+	async rename(newTitle: string, newID?: RoomID, moveLogs = true, keepAliases = true) {
 		if (!newID) newID = toID(newTitle) as RoomID;
-		if (this.game || this.tour) return;
+		if ((this.game && moveLogs) || this.tour) return;
```

Comment:
n.b. ChatRooms can also have games.

Possibly `(this.type === 'chat' && this.game)`.

Also possibly `throw Chat.ErrorMessage` instead of just returning.

Reply:
The intent of the `moveLogs` check is to disallow renaming the logs of battles, since that would disrupt usage stats (or so I'm told), so that needs to stay, but I should check for being a chatroom with an ongoing game.

Reply:
That check could probably happen inside the logs, and it probably needs to, because battles that start hidden also shouldn't have the password in their logs.

Reply:
Where is 'inside the logs'?

Reply:
Inside `server/roomlogs.ts`.

Reply:
Or whatever code handles log writing.

Reply:
Battle logs (which is what we're talking about) seem to be in `server/room-battle.ts`.

Reply:
I wonder if those should be moved... either way, it doesn't seem to need to be handled in `room.rename`. The `moveLogs` parameter appears superfluous.